src/crimson/os/seastore/onode_manager/staged-fltree/node_extent_mutable.h

```diff
@@ -59,20 +54,23 @@ class NodeExtentMutable {
 
   template <typename T>
   void validate_inplace_update(const T& updated) {
-    assert((const char*)&updated >= get_write());
-    assert((const char*)&updated + sizeof(T) <= buf_upper_bound());
+    assert(is_safe(&updated, sizeof(T)));
   }
 
-  const char* get_read() const;
-  char* get_write();
-  extent_len_t get_length() const;
-  laddr_t get_laddr() const;
+  const char* get_read() const { return p_start; }
+  char* get_write() { return p_start; }
+  extent_len_t get_length() const { return length; }
 
  private:
-  explicit NodeExtentMutable(NodeExtent&);
-  const char* buf_upper_bound() const;
+  explicit NodeExtentMutable(char* p_start, extent_len_t length)
```

Comment:
nit, no need to mark the ctor with `explicit` anymore, as it now has two parameters.

Reply:
removed.